redash/models/__init__.py

```diff
@@ -409,6 +410,22 @@ def groups(self):
         return self.data_source.groups
 
 
+@listens_for(BaseQuery, "before_compile", retval=True)
+def prefilter_query_results(query):
+    for desc in query.column_descriptions:
+        if desc['type'] is QueryResult:
```

Comment:
This approach was the best I could come up with. It does ensure that the user only sees the `QueryResult` records that they "own" but doesn't actually use the role / RLS policy on that table because I couldn't figure out a way to actually wrap the query in a `set` / `reset role`, since there is no corresponding "after" event for queries. I'm open to suggestions on how to make it actually use the role, but it's worth noting that the RLS policy on the `redash.query_results` table is still important to prevent the user from issuing queries directly against that table to see results that they shouldn't.

Reply:
Can we add a comment in the code to this affect?